### PR TITLE
Type mapping related improvements and refactoring

### PR DESCRIPTION
**Pull request by hashhar**
## Description

- Added more thorough tests for ARRAY type mapping
- Extracted some modules for mappers and type objects
- Change _create_row_mapper to not depend on specific order of the if else branches

A follow-up PR will come which adds more tests for MAP and ROW types and fixes some bugs that exist there.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

**Comment by hashhar** [comment]
- https://github.com/trinodb/trino-python-client/pull/440/commits/626b2240c31b130b9e431d57fa9f563b51821a6f
- https://github.com/trinodb/trino-python-client/pull/440/commits/60207896e712b5a6114f5b5a1e0d089fcc6b5288
- https://github.com/trinodb/trino-python-client/pull/440/commits/77baefb3b6fbbb57fe0c9c2a7b97f9ed64938808

These are the only functional commits, others are refactors.

**Review by hovaesco — APPROVED** [review]
Looks good, just one comment regarding removed values from map.

File: trino/sqlalchemy/datatype.py [hunk 1]
```diff
@@ -163,11 +163,12 @@ def _format_value(self, value):
...
     #
-    # === Mixed ===
+    # === Others ===
     # 'ipaddress': IPADDRESS
     # 'uuid': UUID,
     # 'hyperloglog': HYPERLOGLOG,
     # 'p4hyperloglog': P4HYPERLOGLOG,
+    # 'setdigest': SETDIGEST,
     # 'qdigest': QDIGEST,
     # 'tdigest': TDIGEST,
```
[the remark this answers is not in the document]
Comment (hashhar) [reply trino/sqlalchemy/datatype.py:173]: Let's still keep those in the map. It's useful to have it to show what types are mapped to what.

Note that some databases for example map their custom types to a custom SQLA type - we can do this too in future if we ever want to.

**Review by ebyhr — APPROVED** [review]
Just skimmed.

**Pull request merged**